docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,188 @@
+- :func:`torch.rand`
+- :func:`torch.randn`
+- :func:`torch.as_strided`
+- :func:`torch.from_numpy`
+- :func:`torch.zeros`
+- :func:`torch.ones`
+- :func:`torch.full`
+- :func:`torch.eye`
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real` view functions:
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+
+The real and imaginary values of a complex tensor can be accessed using the :attr:`real` and
+:attr:`imag` views.
+
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+
+Angle and abs
+-------------
+
+The angle and absolute values of a complex tensor can be accesses using :func:`torch.angle` and
+`torch.abs`.
+
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Linear Algebra
+--------------
+
+Currently, there is very minimal linear algebraic operation support for complex tensors.
+We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
+(the latter three are only supported on CPU). However we are working to add support for more
+functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`, :func:`torch.eig`,
+:func:`torch.symeig`. If any of the other ops would help your use case, please search if an issue has
+already been filed (https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22module%3A+named+tensor%22)
+and if not, file one (https://github.com/pytorch/pytorch/issues/new/choose).
+
+Serialization
+-------------
+
+Complex Tensors can be serialized, allowing data to be saved as complex values.
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+
+Autograd
+--------
+
+PyTorch supports Autograd for Complex Tensors.
+
+1. :func:`torch.functional.backward` can be used for holomorphic :math:`C -> C` functions.
+   For non-holomorphic functions, the gradient is evaluated as if it were holomorphic.
+2. :func:`torch.functional.backward` can be used to optimize :math:`C -> R` functions, like
+   real-values loss functions of complex parameters :math:`x` by taking steps in the direction
+   of conjugate of :math:`x.grad`.
```

Comment:
We might want to double check the conjugate here!
Also if this is true, we might want to update our optimizers to do that when the parameters are complex

Reply:
yeah that's correct! the gradient definition that we are using does specify that right?

Reply:
If it's already in the gradient definition, no need to do it again here.